### ISSUE TITLE
Release management

### ISSUE THREAD

**Issue by kba** [tmbdev/hocr-tools#43]
For one, having users use specific versions makes debugging easier.

The tools could be uploaded to PyPI, so users can install it with `pip install hocr-tools`, or included in distros like Debian.

Possible course of action:
* Release a `0.0.1` version soon (i.e. tag a git commit `v0.0.1`) to have a starting point
* Consider [reorganizing the module](#42)
* Make the tools compatible with PyPI
* Try to adhere to [semantic versioning](http://semver.org/)

The CLI of the tools has not changed or at least not much over the last years. However, this could (and should) change in the future, possibly breaking backwards compatibility if it cannot be avoided.

**Comment by zuphilip** [comment]
What about the[ version in the `setup.py`](https://github.com/tmbdev/hocr-tools/blob/master/setup.py#L7)? Would this then be still needed or how to synchronize?

**Comment by kba** [comment]
Yes, this would have to be synchronized. Maybe start tagging releases with `0.2`, then `0.2.x` for patches etc.

**Comment by zuphilip** [comment]
Okay, I think we should do that and start with creating releases.

On github we can create releases also retrospective for a given commit. I suggest to do
 * https://github.com/tmbdev/hocr-tools/commit/e099ea3697781e6b92f6324a1dfaaf2b26ceb0d4 --> `v0.1` (initial)
 * https://github.com/tmbdev/hocr-tools/commit/4ea25ad64c2db46afa9a654b465bb2f12d1c0e22 --> `v0.2` (hocr-pdf is added)
 * https://github.com/tmbdev/hocr-tools/commit/a627d3ca0b5ec0355a10c2be3b7a39eb802ae5b0 --> `v0.2.1` (bugfixing)
 * https://github.com/tmbdev/hocr-tools/commit/9c1e3174aee4d6498d7d6b8dad4fd918e820b5e6 --> `v0.2.2` (Readme)
 * https://github.com/tmbdev/hocr-tools/commit/cbac14b4a3a5f5d37c73a19f4e6a4dad875ca3f5 --> `v0.3.0` (use lxml, licence, etc.)
 * https://github.com/tmbdev/hocr-tools/commit/e8a2bd35e2f4ed35a08aa7322800ebb5dd3fa519 --> `v0.3.1` (unit tests, Travis)
 * https://github.com/tmbdev/hocr-tools/commit/50c8a9e235566d64891dbe233962c1bdd896951d --> `v0.3.2` (fix rawtext)

What do you think?

**Comment by stweil** [comment]
Was setup.py (added by commit c167b92deb7353d2ce149c0d5b6590aaaea7be51) the first and only reference to something like a release / version? Currently we have only the master branch and no tags at all in git. So there is no version history to consider and we can start freely.
@zuphilip, do you think retrospective tags / releases would be useful? Even if we decide that they are not needed now, it would still be possible to add them later.
IMHO it would be sufficient to start with the latest version, and according to semantic versioning I'd name it v1.0.0.
GitHub supports release downloads from https://github.com/tmbdev/hocr-tools/releases, so we should support that, too (besides PyPI).

**Comment by zuphilip** [comment]
Yes, `setup.py` was added in that commit by @kba and therefore we are free to do with anything (we should just in the end also increase the number there).

Yes, I think also retrospective tags are useful. They are showing the main development steps (see my suggestion above).

Why would you start with `v1...` and not `v0...`?

The downloads on github will be created automatically, see https://github.com/zuphilip/hocr-tools/releases . If we want another format to make the releases available, then this can be added manually or by a script/hook.

**Comment by stweil** [comment]
http://semver.org/#spec-item-4:
> Major version zero (0.y.z) is for initial development. Anything may change at any time. The public API should not be considered stable.

http://semver.org/#spec-item-5:
> Version 1.0.0 defines the public API. The way in which the version number is incremented after this release is dependent on this public API and how it changes.

**Comment by kba** [comment]
:+1:  for starting with `1.0.0`, that makes sense, since the project has been public for a while. Changes like #41 should be a minor version, #20 could be a new major version if they break existing interfaces.

Retrospectively tagging commits is nice to see how the project evolved. Since @zuphilip already researched them and adding the tags is easy :+1:

**Comment by zuphilip** [comment]
Okay, I updated the tags such that they start from 1 and would thereby suggest this https://github.com/zuphilip/hocr-tools/releases , i.e. we would currently be at `v1.2.2`.

Let me know what you think. I would like to do this tomorrow. ⏰

**Comment by kba** [comment]
Looks good to me. I would have released a version `1.0.0` now and tagged the earlier versions like you [proposed earlier](https://github.com/tmbdev/hocr-tools/issues/43#issuecomment-243850646). But it doesn't really matter much as long as it's consistent.

Here's how this will look like in PyPI: https://testpypi.python.org/pypi?:action=display&name=hocr_tools&version=1.2.2 I'll create a new PR for that.

**Comment by zuphilip** [comment]
(The version 1.0.0 from October 2009 was public for 3,5 year until the next commit.)

**Comment by kba** [comment]
Yes, but you could also argue that it was in initial development for a long time (c.f. comments like "we may revisit this issue in later revisions." and the general sparseness/non-existance of the README until recently).

The important thing is that it's consistent in the future.

**Comment by zuphilip** [comment]
Okay, @stweil @kba you convinced me to release now `v1.0.0` and tag the previous history by starting with `v0.1.0`. Therefore, the version 1 is indicating that we are now starting with releasing the tools and tagging them.

**Comment by zuphilip** [comment]
Okay, :sparkles: `v1.0.0` :sparkles: was now released! Thanks to everybody for the help and patience.

Let me know if we can help with the PyPI configuration.
